packages/core/src/internal-types.ts

```diff
@@ -43,3 +44,7 @@ export type PathParameterValues = {[key: string]: any};
 export type OperationArgs = any[];
 export type OperationRetval = any;
 // tslint:enable:no-any
+
+export type getFromContext = (key: string) => BoundValue;
+export type bindElement =
+  (key: string, value: ValueOrPromise<BoundValue>) => void;
```

Comment:
As I commented above, I believe `bindElement` should accept `key` argument only and return the `Binding` instance created, so that users can decide what kind of value (static value, dynamic getter, class, provider) they want to use.